tests/cstor/cspc/positive/provisioning/cspc_provisioning_utils_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The OpenEBS Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	. "github.com/onsi/gomega"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	cspc_v1alpha1 "github.com/openebs/maya/pkg/cstor/poolcluster/v1alpha1"
+	cspcspecs_v1alpha1 "github.com/openebs/maya/pkg/cstor/poolcluster/v1alpha1/cstorpoolspecs"
+	cspcrg_v1alpha1 "github.com/openebs/maya/pkg/cstor/poolcluster/v1alpha1/raidgroups"
+)
+
+func createCSPCObjectForStripe()  {
+	createCSPCObject(1,"stripe")
+}
+
+func createCSPCObjectForMirror()  {
+	createCSPCObject(2,"mirror")
+}
+
+func createCSPCObjectForRaidz()  {
+	createCSPCObject(3,"raidz")
+}
+
+func createCSPCObjectForRaidz2()  {
+	createCSPCObject(6,"raidz2")
+}
+
+func createCSPCObject(blockDeviceCount int,poolType string)  {
+	var err error
+	cspcObj,err = cspc_v1alpha1.NewBuilder().
+		WithGenerateName(cspcName).
+		WithNamespace(ops.NameSpace).
+		WithPoolSpecBuilder(cspcspecs_v1alpha1.NewBuilder().
+			WithNodeSelector(NodeList.Items[0].Labels).
+			WithRaidGroupBuilder(
+				cspcrg_v1alpha1.NewBuilder().
+					WithCSPCBlockDeviceList(ops.GetCSPCBDListForNode(NodeList.Items[0].Name,blockDeviceCount)).
+					WithType(poolType),
+			),
+		).
+		WithPoolSpecBuilder(cspcspecs_v1alpha1.NewBuilder().
+			WithNodeSelector(NodeList.Items[1].Labels).
+			WithRaidGroupBuilder(
+				cspcrg_v1alpha1.NewBuilder().
+					WithCSPCBlockDeviceList(ops.GetCSPCBDListForNode(NodeList.Items[1].Name,blockDeviceCount)).
+					WithType(poolType),
+			),
+		).
+		WithPoolSpecBuilder(cspcspecs_v1alpha1.NewBuilder().
+			WithNodeSelector(NodeList.Items[2].Labels).
+			WithRaidGroupBuilder(
+				cspcrg_v1alpha1.NewBuilder().
+					WithCSPCBlockDeviceList(ops.GetCSPCBDListForNode(NodeList.Items[2].Name,blockDeviceCount)).
+					WithType(poolType),
+			),
+		).
+		GetObj()
+	Expect(err).ShouldNot(HaveOccurred())
+	cspcObj,err = ops.CSPCClient.WithNamespace(ops.NameSpace).Create(cspcObj)
+	Expect(err).To(BeNil())
+
+	Cspc,err = cspc_v1alpha1.BuilderForAPIObject(cspcObj).Build()
+	Expect(err).To(BeNil())
+}
+
+func verifyDesiredCSPICount()  {
+	cspiCount := ops.GetHealthyCSPICount(cspcObj.Name, 3)
```

Comment:
Remove hardcoded values let us take input from cli

Reply:
As discussed, we need to create some way around CSPC object creation so as to run on single-node/ ( or n-node ) cluster, after that this value can be passed via flag. This will be done in a separate PR.